<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MusPlayerQt</class>
 <widget class="QMainWindow" name="MusPlayerQt">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>489</width>
    <height>384</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>250</width>
    <height>0</height>
   </size>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::ContextMenuPolicy::CustomContextMenu</enum>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Simple SDL2 Mixer X Music player</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../_resources/musicplayer.qrc">
    <normaloff>:/cat_musplay/cat_musplay_48x48.png</normaloff>:/cat_musplay/cat_musplay_48x48.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
    background-color: rgb(0, 0, 0);
    color: rgb(0, 255, 0);
}
*:!enabled{
    color: rgb(170, 170, 170);
}
QPushButton {
	background-color: rgb(65, 65, 65);
	border-color: rgb(80, 80, 80);
	border-style: outset;
    border-width: 2px;
	color: rgb(0, 255, 255);
	border-radius: 3px;
    height: 25px;
 }
QPushButton:pressed {
    background-color: rgb(80, 80, 80);
    border-style: inset;
}
QCheckBox {
	border-color: rgb(0, 170, 0);
    border-width: 1px;
	color: rgb(0, 255, 255);
 }
QComboBox, QSpinBox, QLineEdit:enabled{
	border-radius: 3px;
    border-width: 2px;
    border-style: solid;
	border-color: rgb(0, 170, 0);
	color: rgb(0, 255, 255);
}
QSpinBox::up-button {
    subcontrol-origin: border;
	color: rgb(0, 255, 255);
}
QSpinBox::down-button {
    subcontrol-origin: border;
	color: rgb(0, 255, 255);
}
QSpinBox::up-button:!enabled {
    subcontrol-origin: border;
	color: rgb(230, 230, 230);
}
QSpinBox::down-button:!enabled {
    subcontrol-origin: border;
	color: rgb(230, 230, 230);
}


QPushButton:!enabled, QSpinBox:!enabled, QLineEdit:!enabled
{
    border-color: rgb(120, 120, 120);
    color: rgb(128, 128, 128);
}
QGroupBox
{
    padding: 5px;
    border-width: 1px;
    border-style: solid;
    border-color: rgb(0, 170, 0);
	border-radius: 3px;
}
QLabel
{
	background-color: transparent;
}

</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
    </property>
    <item row="3" column="0">
     <widget class="LabelMarquee" name="musAlbum">
      <property name="styleSheet">
       <string notr="true">border-width: 1px;
border-style: solid;
border-color: rgba(0, 0, 0, 0);
border-bottom-color: rgb(0, 120, 0);
</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="LabelMarquee" name="musArtist">
      <property name="styleSheet">
       <string notr="true">border-width: 1px;
border-style: solid;
border-color: rgba(0, 0, 0, 0);
border-bottom-color: rgb(0, 120, 0);
</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Press &quot;Open&quot; or drag&amp;drop music file into this window</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="LabelMarquee" name="musCopyright">
      <property name="styleSheet">
       <string notr="true">border-width: 1px;
border-style: solid;
border-color: rgba(0, 0, 0, 0);
border-bottom-color: rgb(0, 120, 0);
</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QFrame" name="frame_2">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>45</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QPushButton" name="open">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Open</string>
         </property>
         <property name="shortcut">
          <string>Ctrl+O</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="play">
         <property name="minimumSize">
          <size>
           <width>24</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Play</string>
         </property>
         <property name="icon">
          <iconset resource="../_resources/musicplayer.qrc">
           <normaloff>:/buttons/play.png</normaloff>:/buttons/play.png</iconset>
         </property>
         <property name="shortcut">
          <string>Space</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stop">
         <property name="minimumSize">
          <size>
           <width>24</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Stop</string>
         </property>
         <property name="icon">
          <iconset resource="../_resources/musicplayer.qrc">
           <normaloff>:/buttons/stop.png</normaloff>:/buttons/stop.png</iconset>
         </property>
         <property name="shortcut">
          <string>.</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="volume">
         <property name="maximum">
          <number>128</number>
         </property>
         <property name="value">
          <number>128</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="smallInfo">
         <property name="text">
          <string>Info</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="isLooping">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Current music has an embedded loop&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="pixmap">
          <pixmap resource="../_resources/musicplayer.qrc">:/loop_in_use.png</pixmap>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="playingTimeLabel">
         <property name="text">
          <string>--:--:--</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="playingTimeLenghtLabel">
         <property name="text">
          <string>/ --:--:--</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QFrame" name="frame_3">
      <property name="frameShape">
       <enum>QFrame::Shape::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QCheckBox" name="recordWav">
         <property name="text">
          <string>Record WAV</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>148</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QFrame" name="gme_setup">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <property name="horizontalSpacing">
           <number>6</number>
          </property>
          <property name="verticalSpacing">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QPushButton" name="trackPrev">
            <property name="text">
             <string notr="true">&lt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="trackNext">
            <property name="text">
             <string notr="true">&gt;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QSpinBox" name="trackID">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="maximum">
             <number>9999999</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Track number:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QCheckBox" name="disableSpcEcho">
            <property name="toolTip">
             <string>Disable built-in SPC echo</string>
            </property>
            <property name="text">
             <string notr="true"/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="12" column="0">
     <layout class="QHBoxLayout" name="GeneralVolumeFrame">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="toolTip">
         <string>Change the SFX volume (affecting all playing SFX at one time, doesn't affects a music)</string>
        </property>
        <property name="text">
         <string>SFX volume:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="volumeGeneral">
        <property name="toolTip">
         <string>Change the general volume (affecting all playing music and SFX at one time)</string>
        </property>
        <property name="maximum">
         <number>128</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>128</number>
        </property>
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="LabelMarquee" name="musTitle">
      <property name="styleSheet">
       <string notr="true">border-width: 1px;
border-style: solid;
border-color: rgba(0, 0, 0, 0);
border-bottom-color: rgb(0, 120, 0);
</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QFrame" name="tempoFrame">
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="1">
        <widget class="QSlider" name="tempo">
         <property name="toolTip">
          <string>Tempo</string>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TickPosition::TicksAbove</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="tempoLabel">
         <property name="text">
          <string>Tempo (-/+):</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QSlider" name="musicPosition">
      <property name="styleSheet">
       <string notr="true">background-color: rgb(211, 215, 207);</string>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TickPosition::TicksAbove</enum>
      </property>
     </widget>
    </item>
    <item row="11" column="0">
     <widget class="QFrame" name="pitchFrame">
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QGridLayout" name="gridLayout_6">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="pitchLabel">
         <property name="text">
          <string>Pitch (-/+):</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="pitch">
         <property name="toolTip">
          <string>Tempo</string>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TickPosition::TicksAbove</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="QFrame" name="speedFrame">
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="speedLabel">
         <property name="text">
          <string>Speed (-/+):</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="speed">
         <property name="toolTip">
          <string>Tempo</string>
         </property>
         <property name="minimum">
          <number>-100</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TickPosition::TicksAbove</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QFrame" name="gainFrame">
      <property name="frameShape">
       <enum>QFrame::Shape::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="gainLabel">
         <property name="text">
          <string>Gain (-/+):</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSlider" name="gain">
         <property name="toolTip">
          <string>Tempo</string>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TickPosition::TicksAbove</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>489</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuSetup">
    <property name="title">
     <string>Setup</string>
    </property>
    <widget class="QMenu" name="menuMusicLooping">
     <property name="title">
      <string>Music looping</string>
     </property>
     <addaction name="actionLoopForever"/>
     <addaction name="separator"/>
     <addaction name="actionPlay1Time"/>
     <addaction name="actionPlay2Times"/>
     <addaction name="actionPlay3Times"/>
    </widget>
    <addaction name="actionAudioSetup"/>
    <addaction name="actionMidiSetup"/>
    <addaction name="actionMultiMusicTesting"/>
    <addaction name="actionMultiSFXTesting"/>
    <addaction name="actionSfxTesting"/>
    <addaction name="actionMusicFX"/>
    <addaction name="actionTracksMuter"/>
    <addaction name="actionFileAssoc"/>
    <addaction name="separator"/>
    <addaction name="actionEnableReverb"/>
    <addaction name="actionTuneReverb"/>
    <addaction name="actionEnableEcho"/>
    <addaction name="actionTuneEcho"/>
    <addaction name="separator"/>
    <addaction name="menuMusicLooping"/>
    <addaction name="actionPause_the_audio_stream"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelpGitHubMixerX"/>
    <addaction name="actionHelpGitHub"/>
    <addaction name="actionHelpLicense"/>
    <addaction name="separator"/>
    <addaction name="actionHelpAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSetup"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionMidiSetup">
   <property name="text">
    <string>MIDI Setup</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionHelpLicense">
   <property name="text">
    <string>GPL License</string>
   </property>
  </action>
  <action name="actionHelpAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionHelpGitHub">
   <property name="text">
    <string>Visit Github page of Moondust Project...</string>
   </property>
  </action>
  <action name="actionSfxTesting">
   <property name="text">
    <string>SFX Testing</string>
   </property>
  </action>
  <action name="actionEnableReverb">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Enable reverb</string>
   </property>
  </action>
  <action name="actionFileAssoc">
   <property name="text">
    <string>File associations...</string>
   </property>
  </action>
  <action name="actionLoopForever">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Loop forever</string>
   </property>
  </action>
  <action name="actionPlay1Time">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Play 1 time</string>
   </property>
  </action>
  <action name="actionPlay2Times">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Play 2 times</string>
   </property>
  </action>
  <action name="actionPlay3Times">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Play 3 times</string>
   </property>
  </action>
  <action name="actionAudioSetup">
   <property name="text">
    <string>Audio Setup...</string>
   </property>
  </action>
  <action name="actionEnableEcho">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Enable echo</string>
   </property>
  </action>
  <action name="actionTuneEcho">
   <property name="text">
    <string>Tune echo...</string>
   </property>
  </action>
  <action name="actionMultiMusicTesting">
   <property name="text">
    <string>Multi-Music Testing</string>
   </property>
  </action>
  <action name="actionMusicFX">
   <property name="text">
    <string>Music FX</string>
   </property>
  </action>
  <action name="actionTracksMuter">
   <property name="text">
    <string>Tracks enable/disable</string>
   </property>
  </action>
  <action name="actionPause_the_audio_stream">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause the audio stream</string>
   </property>
  </action>
  <action name="actionTuneReverb">
   <property name="text">
    <string>Tune reverb...</string>
   </property>
  </action>
  <action name="actionMultiSFXTesting">
   <property name="text">
    <string>Multi-SFX testing</string>
   </property>
  </action>
  <action name="actionHelpGitHubMixerX">
   <property name="text">
    <string>Visit GitHub page of MixerX...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>LabelMarquee</class>
   <extends>QLabel</extends>
   <header>MainWindow/label_marquee.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../_resources/musicplayer.qrc"/>
 </resources>
 <connections/>
</ui>
